exercises/colorsquare.js

```diff
@@ -24,10 +31,14 @@ class ColorSquare{
 		true / false, depending on success
 	notes:
 		make sure it only sets the neighbor if it is the right class constructor (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/constructor)
-		if it is the right constructor, set the neightbor
+		if it is the right constructor, set the neighbor
 	*/
-	set neighbor(){
-
+	set neighbor(newNeighbor){
+        if (newNeighbor){
+            if (newNeighbor.constructor.name === "ColorSquare") {
```

Comment:
While this works, you don't have to test versus the name, as there might conceivably been two different objects.

One way to do it is this:

```if( newNeighbor.constructor === ColorSquare ) ```

However an even better way is this:

``` if(newNeighbor.constructor === this.constructor ) ```